exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -0,0 +1,416 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+
+/*
+ * Prometheus instrumentation library for JVM applications
+ * Copyright 2012-2015 The Prometheus Authors
+ *
+ * This product includes software developed at
+ * Boxever Ltd. (http://www.boxever.com/).
+ *
+ * This product includes software developed at
+ * SoundCloud Ltd. (http://soundcloud.com/).
+ *
+ * This product includes software developed as part of the
+ * Ocelli project by Netflix Inc. (https://github.com/Netflix/ocelli/).
+ */
+
+package io.opentelemetry.exporter.prometheus;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.metrics.data.DoubleHistogramPointData;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import io.opentelemetry.sdk.metrics.data.DoubleSummaryPointData;
+import io.opentelemetry.sdk.metrics.data.ExemplarData;
+import io.opentelemetry.sdk.metrics.data.LongPointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.PointData;
+import io.opentelemetry.sdk.metrics.data.ValueAtPercentile;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.regex.Pattern;
+
+/** Serializes metrics into Prometheus exposition formats. */
+// Adapted from
+// https://github.com/prometheus/client_java/blob/master/simpleclient_common/src/main/java/io/prometheus/client/exporter/common/TextFormat.java
+abstract class Serializer {
+
+  private static final Pattern SANITIZE_PREFIX_PATTERN = Pattern.compile("^[^a-zA-Z_:]");
+  private static final Pattern SANITIZE_BODY_PATTERN = Pattern.compile("[^a-zA-Z0-9_:]");
+
+  abstract String headerName(String name, PrometheusType type);
+
+  abstract void writeHelp(Writer writer, String description) throws IOException;
+
+  abstract void writeTimestamp(Writer writer, long timestampNanos) throws IOException;
+
+  abstract void writeExemplar(
+      Writer writer, Collection<ExemplarData> exemplars, double minExemplar, double maxExemplar)
+      throws IOException;
+
+  abstract void writeEof(Writer writer) throws IOException;
+
+  final void write(Collection<MetricData> metrics, OutputStream output) throws IOException {
+    try (Writer writer =
+        new BufferedWriter(new OutputStreamWriter(output, StandardCharsets.UTF_8))) {
+      for (MetricData metric : metrics) {
+        write(metric, writer);
+      }
+      writeEof(writer);
+    }
+  }
+
+  private void write(MetricData metric, Writer writer) throws IOException {
+    // TODO: Implement
```

Comment:
Can you open a bug for this one?

Reply:
I suspect this would be an issue in the spec repo. I changed this comment a bit to reflect this proposed spec wording

https://github.com/open-telemetry/opentelemetry-specification/pull/2266/files#diff-0efae13f08f98e62a81767d5daeff37ebb7ef8c50537c7b9013e72506e9b055aR1152